parcels/kernel.py

```diff
@@ -445,9 +445,7 @@ def __add__(self, kernel):
         return self.merge(kernel, type(self))
 
     def __radd__(self, kernel):
-        if not isinstance(kernel, type(self)):
-            kernel = type(self)(self.fieldset, self.ptype, pyfunc=kernel)
-        return kernel.merge(self, type(self))
+        return self + kernel
```

Comment:
No need for code duplication here. Just fallback to the `__add__` method as its commutative

[More info on `__radd__`](https://mathspp.com/blog/pydonts/overloading-arithmetic-operators-with-dunder-methods#reflected-dunder-methods)